src/Api/Mastodon/Account.php

```diff
@@ -52,37 +53,52 @@ class Account
 
 	/**
-	 * Creates an account record from a contact record. Expects all contact table fields to be set
+	 * Creates an account record from a public contact record. Expects all contact table fields to be set.
 	 *
-	 * @param array $contact   Full contact table record
-	 * @param array $apcontact Full apcontact table record
+	 * @param BaseURL $baseUrl
+	 * @param array   $publicContact Full contact table record with uid = 0
+	 * @param array   $apcontact     Optional full apcontact table record
 	 * @return Account
 	 * @throws \Friendica\Network\HTTPException\InternalServerErrorException
 	 */
-	public static function createFromContact(array $contact, array $apcontact = [])
+	public static function create(BaseURL $baseUrl, array $publicContact, array $apcontact = [])
 	{
 		$account = new Account();
-		$account->id              = $contact['id'];
-		$account->username        = $contact['nick'];
-		$account->acct            = $contact['nick'];
-		$account->display_name    = $contact['name'];
+		$account->id              = $publicContact['id'];
+		$account->username        = $publicContact['nick'];
+		$account->acct            =
+			strpos($publicContact['url'], $baseUrl->get() . '/') === 0 ?
+			$publicContact['nick'] :
+			$publicContact['addr'];
+		$account->display_name    = $publicContact['name'];
 		$account->locked          = !empty($apcontact['manually-approve']);
-		$account->created_at      = DateTimeFormat::utc($contact['created'], DateTimeFormat::ATOM);
+		$account->created_at      = DateTimeFormat::utc($publicContact['created'], DateTimeFormat::ATOM);
 		$account->followers_count = $apcontact['followers_count'] ?? 0;
 		$account->following_count = $apcontact['following_count'] ?? 0;
 		$account->statuses_count  = $apcontact['statuses_count'] ?? 0;
-		$account->note            = BBCode::convert($contact['about'], false);
-		$account->url             = $contact['url'];
-		$account->avatar          = $contact['avatar'];
-		$account->avatar_static   = $contact['avatar'];
+		$account->note            = BBCode::convert($publicContact['about'], false);
+		$account->url             = $publicContact['url'];
+		$account->avatar          = $publicContact['avatar'];
+		$account->avatar_static   = $publicContact['avatar'];
 		// No header picture in Friendica
 		$account->header          = '';
 		$account->header_static   = '';
 		// No custom emojis per account in Friendica
 		$account->emojis          = [];
-		$account->bot             = ($contact['contact-type'] == Contact::TYPE_NEWS);
+		// No metadata fields in Friendica
+		$account->fields          = [];
+		$account->bot             = ($publicContact['contact-type'] == Contact::TYPE_NEWS);
+		$account->group           = ($publicContact['contact-type'] == Contact::TYPE_COMMUNITY);
+		$account->discoverable    = !$publicContact['unsearchable'];
+		$account->last_status_at  = !empty($publicContact['last-item']) ? DateTimeFormat::utc($publicContact['last-item'], DateTimeFormat::ATOM) : null;
```

Comment:
For the "last-item" we need both arrays, "contact" and "publiccontact". When you subscribed to someone who mostly posts private posts, then the "contact" will have newer entries.

Reply:
Accounts entities are showing public information and as such shouldn’t show the last email a mail contact sent. Public feeds should have a public contact and their last item date should be updated when it’s polled for anyone following it.

Reply:
When a specific user is calling that API, then the specific user should get the correct information.

Reply:
Hum, you’re right.

Reply:
I added the user contact data superseding, but it isn't useful in either of the current cases (no private interactions with introductions and the instance contact is public info).

Reply:
When you are logged in, you should get the right data - even with public endpoints.